datapackage_pipelines_measure/processors/add_twitter_resource.py

```diff
@@ -0,0 +1,285 @@
+    user_tweets_args = {'screen_name': entity, 'count': 200,
+                        'include_rts': False, 'exclude_replies': False}
+    start_date = datetime. \
+        datetime.strptime(start_date, TWITTER_API_DATE_RANGE_FORMAT).date()
+    end_date = datetime. \
+        datetime.strptime(end_date, TWITTER_API_DATE_RANGE_FORMAT).date()
+    api = _get_twitter_api_handler()
+
+    for tweet in _handle_twitter_rate_limit(tweepy_cursor(
+            api.user_timeline, **user_tweets_args).items()):
+        if tweet.created_at.date() >= end_date:
+            continue
+        if tweet.created_at.date() < start_date:
+            break
+        all_self_tweets_in_time_period.append(tweet)
+    return all_self_tweets_in_time_period
+
+
+def _get_mentions_for_entity_from_source(entity, start_date, end_date):
+    '''Get the mentions metric for entity type.'''
+    all_tweets_in_search = _get_twitter_search_results(entity,
+                                                       start_date,
+                                                       end_date)
+    return len(all_tweets_in_search)
+
+
+def _get_interactions_for_entity_from_source(entity, start_date, end_date):
+    def _count_indicators_of_interaction(list_of_tweets):
+        '''This methods iterates over a list of tweets, and returns a two-tuple
+        of total count of favorites, and total count of retweets. The method
+        makes sure it doesn't double-count stats, so it skips tweets that are
+        themselves retweets, and counts only original tweets'''
+        favorite_count, retweet_count = 0, 0
+        for tweet in list_of_tweets:
+            if getattr(tweet, 'retweeted_status', None):
+                continue
+            favorite_count += tweet.favorite_count
+            retweet_count += tweet.retweet_count
+        return favorite_count, retweet_count
+
+    entity_type = _get_entity_type(entity)
+    if entity_type == 'account':
+        '''
+        Get interaction for a twitter account entity. This is done by getting
+        all the account tweets during given date range, and aggregate the
+        measured 'interaction' metrics - favorites and retweets.
+        '''
+        tweets = _get_account_tweets(entity, start_date, end_date)
+    elif entity_type == 'hashtag':
+        '''
+        Get interactions for a hashtag entity. This is done by searching for
+        the hashtag between the given date range, and aggregate the measured
+        'interaction' metrics - favorites and retweets.
+        '''
+        tweets = _get_twitter_search_results(entity, start_date, end_date)
+    favorite_count, retweet_count = _count_indicators_of_interaction(tweets)
+    return sum([favorite_count, retweet_count])
+
+
+parameters, datapackage, res_iter = ingest()
+
+entity = parameters['entity']
+project_id = parameters['project_id']
+safe_entity = _get_safe_entity(entity)
+resource = {
+    'name': safe_entity,
+    'path': 'data/{}.json'.format(safe_entity)
```

Comment:
Again, this should be a `.csv`. Though, most of the time this path is never used, as the resources are concatenated later in the pipeline.